### DISCUSSION TITLE
FileLoad Error for .datx ActivPal files

### DISCUSSION THREAD

**Discussion by JanisFie** [Ergo-Tools/ActiPASS#96] [Crashes, errors and bugs]
Hi,

we have used activPal to assess Physical Behavior in a nationwide sample of children and adolescents.
When I try to process the data with ActiPASS default settings, without diary and calibration, I get the following error:
"FileLoad: Error reading file: The sample rate is outside the excepted range in file"
The file can be processed using the same file in .csv format. The sample rate should be 20 Hz.

I also tested various files and get the same issue with each. .datx doesn't work, .csv does. Since we extracted all files (3k+) as .datx it would be very convenient if there was a solution.

You can find the files in the email @pasan-h 
I am aware that you cannot provide individual support, but this seems like a general issue.

Best,
Janis

**Comment by pasan-h** [comment]
Hi Janis,
Please use “Discussions” to report problems/bugs in general. The “Issues” should be used when you can trace the a bug to a specific source file. We use this distinction to make sure all users benefit from reported problems.
I already moved your issue to correct discussion category.

See guideline for how to reporting issues.
https://github.com/Ergo-Tools/ActiPASS/discussions/56

Can you add the relevant info to your post, then I can look into it as time permits. Anyway when I processed the file you attached, it worked ok for me. See attached activity classification.
BR
/Pasan

![quality_check](https://github.com/user-attachments/assets/f2746ec9-243c-45c1-aaa2-15eb29fa8a51)

**Reply by JanisFie** [reply]
Thank you for the swift response. 
The relevant info are:
ActiPASS Version:	2024.06.1
system information: Windows 10 Version 22H2 (Build 19045.5011)
There is no crash_log but I attached the BatchLog and BatchSettings (with a modified file path):
[ActiPASS_BatchLog.txt](https://github.com/user-attachments/files/17716586/ActiPASS_BatchLog.txt)
[ActiPASS_BatchSettings.txt](https://github.com/user-attachments/files/17716621/ActiPASS_BatchSettings.txt)

The .csv file can be processed, the .datx file produces the error above.
Let me know if you need any additional information.

**Comment by claaslendt** [comment]
Potentially a long shot, but you may try setting the Windows Regional Format to English @JanisFie.

@pasan-h I cannot find the loading function for .datx files on GitHub?

**Reply by pasan-h** [reply]
> The .csv file can be processed, the .datx file produces the error above.

Ok, now I understand. Does the *.datx files come from ActivPAL 4 devices. Then ActiPASS cannot open them properly. ActivPAL datx file format is proprietary and there are no open code to read datx files from newer version 4 or later devices. ActiPASS uses this library to open datx files from version 3 devices.
https://github.com/R-Broadley/activpal_utils-matlab

If the datx file is from ActivPAL version 3 device, also try the fix @claaslendt mentioned above.

For file format support please see:
https://github.com/Ergo-Tools/ActiPASS/wiki/ActiPASS-User-Guide#start-a-new-batch-analysis

I agree that the error produced is not very informative and kind of misleading. ActiPASS just prints the error from above library.

**Reply by JanisFie** [reply]
That explains the issue thanks! I missed that ActivPal 4 .datx is not possible. Thank you for your time!
